### PR TITLE
Add Basic Leader entity

### PR DESCRIPTION
**Pull request by DeepankarGupta13**
Added Basic Leadear.
Currently we can only add vertices, arrowHead, pathType.
need to add MLEADER entity and Text in Leader.

### CODE REVIEW

File: src/Sections/EntitiesSection/Entities/Leader.ts [hunk 1]
```diff
@@ -0,0 +1,52 @@
+import Entity, { CommonEntityOptions } from '../Entity';
+import { BoundingBox, boundingBox_t } from 'Internals/BoundingBox';
+import { Dxfier } from 'Internals/Dxfier';
+import { vec2_t, point3d } from 'Internals/Helpers';
+
+export enum ArrowHeadFlags {
+    ArrowHeadDisabed = 0,
+    ArrowHeadEnabled = 1,
+}
```
Comment (tarikjabiri) [inline src/Sections/EntitiesSection/Entities/Leader.ts:9]: Use it like this:
```ts
export enum ArrowHeadFlags {
    Disabed = 0,
    Enabled = 1,
}
```

**Review by tarikjabiri — CHANGES_REQUESTED** [review]

File: src/Sections/EntitiesSection/Entities/Leader.ts [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+export enum PathType {
+    StarightLine = 0,
```
[the remark this answers is not in the document]
Comment (DeepankarGupta13) [reply src/Sections/EntitiesSection/Entities/Leader.ts]: done sir

File: src/Sections/EntitiesSection/Entities/Leader.ts [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+export class Leader extends Entity {
+    flags: ArrowHeadFlags;
+    pathType: PathType;
+    vertices: LeaderVertex[];
+
+    public constructor(vertices: LeaderVertex[], options?: LeaderOptions) {
+        super('LEADER', 'AcDbLeader', options);
+        this.vertices = vertices;
+        this.flags = options?.flags ?? ArrowHeadFlags.ArrowHeadEnabled;
+        this.pathType = options?.pathType ?? PathType.StarightLine;
```
[the remark this answers is not in the document]
Comment (DeepankarGupta13) [reply src/Sections/EntitiesSection/Entities/Leader.ts]: done sir

File: src/Sections/EntitiesSection/Entities/Leader.ts [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+    }
+
+    override boundingBox(): boundingBox_t {
+        return BoundingBox.verticesBBox(
+            this.vertices.map((vertex) => point3d(vertex.point.x, vertex.point.y, 0))
```
[the remark this answers is not in the document]
Comment (DeepankarGupta13) [reply src/Sections/EntitiesSection/Entities/Leader.ts]: done sir

File: src/Sections/EntitiesSection/Entities/Leader.ts (hunk 1 above)
Comment (DeepankarGupta13) [reply src/Sections/EntitiesSection/Entities/Leader.ts]: done sir

File: src/Sections/EntitiesSection/Entities/Leader.ts [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+}
+
+export enum PathType {
+    StraightLine = 0,
+    Spline = 1,
+}
+
+export interface LeaderVertex {
+    point: vec2_t;
+}
```
Comment (tarikjabiri) [inline src/Sections/EntitiesSection/Entities/Leader.ts:18]: The leader vertex its only a position so no need for a new structure, and also can have a z component so why not use an array of vec3_t:
```ts
public constructor(vertices: vec3_t[], options?: LeaderOptions)
```
and in dxfy:

```ts
for (const v of this.vertices) {
    dx.point3d(v);
}
```

File: src/Sections/EntitiesSection/Entities/Leader.ts [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+
+export interface LeaderOptions extends CommonEntityOptions {
+    flags?: ArrowHeadFlags;
```
Comment (tarikjabiri) [inline src/Sections/EntitiesSection/Entities/Leader.ts:21]: its only a flag not flags, I just looked in the spec they call it **Arrowhead flag**

File: src/Sections/EntitiesSection/Entities/Leader.ts [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+import { BoundingBox, boundingBox_t } from 'Internals/BoundingBox';
+import { Dxfier } from 'Internals/Dxfier';
+import { vec2_t, point3d } from 'Internals/Helpers';
+
+export enum ArrowHeadFlags {
+    Disabed = 0,
+    Enabled = 1,
+}
+
+export enum PathType {
```
Comment (tarikjabiri) [inline src/Sections/EntitiesSection/Entities/Leader.ts:11]: I prefer to call this `LeaderPathType`

**Review by tarikjabiri — CHANGES_REQUESTED** [review]

**Comment by DeepankarGupta13** [comment]
Sir, how to resolve these merge conflicts, I am not able to see then in vs code?

**Comment by tarikjabiri** [comment]
You have this confilcts because you didn't use the latest main branch updates.

**Comment by tarikjabiri** [comment]
You didn't sync you repo
conflicts

File: src/Sections/EntitiesSection/Entities/Leader.ts (hunk 5 above)
Comment (DeepankarGupta13) [reply src/Sections/EntitiesSection/Entities/Leader.ts:18]: done sir

**Comment by DeepankarGupta13** [comment]
sir, pls check once, I have resolved the merge conflicts.

**Comment by tarikjabiri** [comment]
@DeepankarGupta13 did you test it in AutoCAD, it seems not working 🤔.

**Comment by DeepankarGupta13** [comment]
ok sir, i'll check again.

**Comment by DeepankarGupta13** [comment]
it's working by below code:

const vertices = [
    point3d(0, 0, 0),
    point3d(1, 1, 0),
];
const leader = dxf.addLeader(vertices);

**Comment by DeepankarGupta13** [comment]
image

**Comment by DeepankarGupta13** [comment]
pls check again, sir.

**Review by tarikjabiri — APPROVED** [review]

**Comment by DeepankarGupta13** [comment]
Did it worked?

**Comment by tarikjabiri** [comment]
No

**Comment by DeepankarGupta13** [comment]
[example.txt](https://github.com/dxfjs/writer/files/10053816/example.txt)

sir try the above txt file's content. see if that works on 2013

**Comment by tarikjabiri** [comment]
Yeah this one is working 🎉

**Comment by tarikjabiri** [comment]
So the problem was the dimension style name.

**Pull request merged**